react/ProductCustomizer.js

```diff
@@ -1,60 +1,344 @@
+  }
+
+  /**
+  * onHandleNumericStepperChange
+  * Sets the optional variation values by index.
+  * @param object variationObject
+  * @return void
+  */
+  onHandleNumericStepperChange = variationObject => {
+    const {
+      choosedAmount,
+    } = this.state
+
+    choosedAmount[variationObject.index] = variationObject.quantity
+
+    this.setState({ choosedAmount })
+  }
+
+  /**
+  * handleSelectedVariation
+  * Resets extra variations state and update the current variation selected.
+  * @param object variationObject
+  * @return void
+  */
+  handleSelectedVariation = variationObject => {
+    this.setState({ extraVariations: [] })
+    this.setState({
+      selectedVariation: {
+        skuId: variationObject.skuId,
+        variation: variationObject.variation,
+        quantity: variationObject.quantity,
+      },
+    })
+  }
+
+  /**
+  * handleSelectedExtraVariations
+  * Add the changed optional variation and call the calculate method
+  * @param object variationObject
+  * @return void
+  */
+  handleSelectedExtraVariations = variationObject => {
+    const currentExtraVariations = this.state.extraVariations
+    const key = currentExtraVariations.findIndex(extraVariation => {
+      return extraVariation.index === variationObject.index
+    })
+
+    this.onHandleNumericStepperChange(variationObject)
+
+    if (key === -1) {
+      currentExtraVariations.push(variationObject)
+    } else {
+      if (variationObject.quantity !== 0) {
+        currentExtraVariations[key] = variationObject
+      } else {
+        currentExtraVariations.splice(key)
+      }
+    }
+
+    this.setState({ extraVariations: currentExtraVariations }, () => {
+      this.calculateTotalFromSelectedVariation()
+    })
+  }
+
+  /**
+  * calculateTotalFromSelectedVariation
+  * Calculates the total based on all items selected.
+  * @return void
+  */
+  calculateTotalFromSelectedVariation = () => {
+    const {
+      extraVariations,
+      selectedVariation: {
+        quantity,
+        variation,
+      },
+    } = this.state
+
+    const totalVariation = (variation.price / 100) * quantity
+    const totalExtraVariations = extraVariations.reduce((accumulator, item) => {
+      const parsedPrice = parseFloat(item.variation.price / 100).toFixed(2)
+      return accumulator + parsedPrice * item.quantity
+    }, 0)
+
+    this.setState({ total: totalVariation + totalExtraVariations })
+  }
+
+  /**
+  * handleOnSubmitForm
+  * Create an object based on selected variations and send to Order Form.
+  * @return void
+  */
+  handleOnSubmitForm = e => {
+    e.preventDefault()
+
+    const {
+      orderFormContext,
+    } = this.props
+
+    const {
+      selectedVariation,
+    } = this.state
+
+    const service = new ProductCustomizerService()
+    const attachmentsStringsObject = service.createAttachmentStringBySelections(this.state)
+
+    console.log('TO-DO: Insert these strings into a OrderForm', attachmentsStringsObject)
```

Comment:
Remove `console.log`. If this is a pending task then just add a comment with a TODO at the beginning of it.